### PR TITLE
Refactoring2

### PR DESCRIPTION
**Pull request by Amir-Gaifullin**

### CODE REVIEW

File: app/interactors/update_project/save_record.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class UpdateProject
   class SaveRecord
     include Interactor
 
-    delegate :prepared_project_params, to: :context
+    delegate :prepared_project_params, :project to: :context
```
Comment (DmitryBarskov) [inline app/interactors/update_project/save_record.rb]: This won't even work, it is a syntax error

File: app/interactors/update_project/save_record.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def call
     private
 
     def project
-      @project ||= Project.update(prepared_project_params)
+      @project ||= Project.find(project.id)
```
Comment (DmitryBarskov) [inline app/interactors/update_project/save_record.rb]: You don't need to find the project if it is already in the context

**Review by DmitryBarskov — CHANGES_REQUESTED** [review]

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -37,7 +36,7 @@ def create
 
   # PATCH/PUT /projects/1
   def update
-    if @project.update(project_params)
+    if @project.update_project.project
```
Comment (KilinMikhail) [inline app/controllers/projects_controller.rb]:
```suggestion
    if update_project.project
```

File: app/interactors/update_project.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class UpdateProject
+	include Interactor::Organizer
+
+	organize UpdateProject::PrepareParams,
+           UpdateProject::SaveRecord,
+           # UpdateProject::SendNotifications
```
Comment (KilinMikhail) [inline app/interactors/update_project.rb]: why commented?
And "," in the line before is a syntax error in this case

Comment (Amir-Gaifullin) [reply app/interactors/update_project.rb]: i dont save change in this file, my bad

**Comment by Amir-Gaifullin** [comment]
@KilinMikhail I fixed code

**Review by KilinMikhail — APPROVED** [review]
Looks good for me

**Pull request closed without merging**